src/modules/decklink/util/util.h

```diff
@@ -252,9 +252,17 @@ static com_ptr<IDeckLink> get_device(size_t device_index)
 template <typename T>
 static std::wstring get_model_name(const T& device)
 {
+
```

Comment:
Why was this change needed?
It was much cleaner before without any ifdefs, and worked as far as Im aware.

Reply:
Because different string types in Windows/Linux/Apple
1. https://forum.blackmagicdesign.com/viewtopic.php?p=423675#p423675
2. https://www.viva64.com/en/w/v745/print
3. Like this - https://github.com/Streampunk/macadam/blob/master/src/macadam.cc

Reply:
Yes, but we don't need to do it via ifdefs in this way.
`String` is a custom typedef in https://github.com/CasparCG/server/blob/master/src/modules/decklink/decklink_api.h that on windows is `BSTR` and on linux is `const char*`

I am assuming there was some static analysis problem here due to BSTR being more than just a byte array and it being treated as such when passed into `u16(`, so how about adding a `std::wstring to_string(String);` for each platform instead in decklink_api.h which this code can use instead of u16?

Reply:
It's best idea.